src/libostree/ostree-repo.c

```diff
@@ -2271,6 +2783,38 @@ reload_core_config (OstreeRepo          *self,
       return FALSE;
   }
 
+  { g_autofree char *lock_timeout_str = NULL;
+    g_autoptr(GError) local_error = NULL;
+
+    /* Default is set at repo_init() time so that the timeout can be
+     * configured by ostree programs without being overwritten when the
+     * configuration is reloaded.
+     */
+    lock_timeout_str = g_key_file_get_value (self->config, "core", "lock-timeout",
+                                             &local_error);
+    if (lock_timeout_str == NULL)
+      {
+        if (g_error_matches (local_error, G_KEY_FILE_ERROR,
+                             G_KEY_FILE_ERROR_KEY_NOT_FOUND))
+          g_clear_error (&local_error);
+        else
+          {
+            g_propagate_error (error, g_steal_pointer (&local_error));
+            return FALSE;
+          }
+      }
+    else
+      {
+        char *endptr = NULL;
+        gint64 lock_timeout = g_ascii_strtoll (lock_timeout_str, &endptr, 0);
+        if ((lock_timeout < -1) ||
+            (lock_timeout > G_MAXINT) ||
+            (lock_timeout == 0 && endptr == lock_timeout_str))
```

Comment:
This doesn’t check for the case where `*endptr != '\0'`, which could be caused by `lock_timeout_str == "123somestring"`.

Reply:
Good catch. I hadn't thought of that.

Reply:
The `g_ascii_strto*()` are an absolute nightmare for correct error handling.

GLib recently acquired `g_ascii_string_to_{un,}signed()`, which are a lot better, but unfortunately too recent for libostree to use yet. :-(

Reply:
Will fix after #1343. This part didn't make it into the slimmed down setup.